fetcher.go

```diff
@@ -1,37 +1,19 @@
 package fetcher
 
 import (
-	"bytes"
 	"context"
-	"fmt"
-	"io"
 
-	"github.com/ipfs/go-blockservice"
-	dagpb "github.com/ipld/go-codec-dagpb"
 	"github.com/ipld/go-ipld-prime"
-	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
-	basicnode "github.com/ipld/go-ipld-prime/node/basic"
-	"github.com/ipld/go-ipld-prime/schema"
-	"github.com/ipld/go-ipld-prime/traversal"
-	"github.com/ipld/go-ipld-prime/traversal/selector"
-	"github.com/ipld/go-ipld-prime/traversal/selector/builder"
 )
 
-// FetcherConfig defines a configuration object from which Fetcher instances are constructed
-type FetcherConfig struct {
-	blockService     blockservice.BlockService
-	NodeReifier      ipld.NodeReifier
-	PrototypeChooser traversal.LinkTargetNodePrototypeChooser
-}
-
 // Fetcher is an interface for reading from a dag. Reads may be local or remote, and may employ data exchange
 // protocols like graphsync and bitswap
 type Fetcher interface {
 	// NodeMatching traverses a node graph starting with the provided node using the given selector and possibly crossing
 	// block boundaries. Each matched node is passed as FetchResult to the callback. Errors returned from callback will
 	// halt the traversal. The sequence of events is: NodeMatching begins, the callback is called zero or more times
 	// with a FetchResult, then NodeMatching returns.
-	NodeMatching(context.Context, ipld.Node, selector.Selector, FetchCallback) error
+	NodeMatching(context.Context, ipld.Node, ipld.Node, FetchCallback) error
```

Comment:
this isn't just any node though, it's still a selector? is there a way to make that signature clearer while still being efficient @warpfork? 

At least we probably want a comment on it / usage example

Reply:
technically though it is any ipld.Node that parses to a selector.

Reply:
i think what I'm noting is that this interface method signature now provides no indication of which argument is the root and which is the selector. that needs to exist as argument names / comment if it isn't clear from the type signature.

Reply:
yea totally :)

Reply:
updated somewhat, hopefully in a way thats clear